docs/data-ai/train/create-dataset.md

```diff
@@ -0,0 +1,1313 @@
+    );
+
+    viamClient.disconnect();
+    return 0;
+  } catch (error) {
+    console.log("Error creating dataset. It may already exist.");
+    console.log("See: https://app.viam.com/data/datasets");
+    console.log(`Exception: ${error}`);
+    viamClient.disconnect();
+    return 1;
+  }
+}
+
+if (require.main === module) {
+  main().then((code) => process.exit(code));
+}
+```
+
+{{% /tab %}}
+{{% tab name="Flutter" %}}
+
+```dart
+import 'package:viam_sdk/viam_sdk.dart';
+
+// Configuration constants - replace with your actual values
+const String datasetName = ""; // a unique, new name for the dataset you want to create
+const String orgId = ""; // your organization ID, find in your organization settings
+const String partId = ""; // ID of machine that captured target images, find in machine config
+const String apiKey = ""; // API key, find or create in your organization settings
+const String apiKeyId = ""; // API key ID, find or create in your organization settings
+const int maxMatches = 500;
+
+Future<ViamClient> connect() async {
+  return await ViamClient.withApiKey(
+    apiKey: apiKey,
+    apiKeyId: apiKeyId,
+  );
+}
+
+Future<List<BinaryData>> fetchBinaryDataIds(
+    DataClient dataClient, String partId) async {
+  final filter = Filter(partId: partId);
+  final List<BinaryData> allMatches = [];
+  String? last;
+
+  print("Getting data for part...");
+
+  while (allMatches.length < maxMatches) {
+    print("Fetching more data...");
+
+    final response = await dataClient.binaryDataByFilter(
+      filter: filter,
+      limit: 50,
+      last: last,
+      includeBinaryData: false,
+    );
+
+    if (response.data.isEmpty) {
+      break;
+    }
+
+    allMatches.addAll(response.data);
+    last = response.last;
+  }
+
+  return allMatches;
+}
+
+Future<int> main() async {
+  final viamClient = await connect();
+  final dataClient = viamClient.dataClient;
+
+  final matchingData = await fetchBinaryDataIds(dataClient, partId);
+
+  print("Creating dataset...");
+
+  try {
+    final datasetId = await dataClient.createDataset(
+      name: datasetName,
+      organizationId: orgId,
+    );
+    print("Created dataset: $datasetId");
+
+    print("Adding data to dataset...");
+
+    final binaryIds = matchingData
+        .map((obj) => obj.metadata.binaryDataId)
+        .toList();
+
+    await dataClient.addBinaryDataToDatasetByIds(
+      binaryIds: binaryIds,
+      datasetId: datasetId,
+    );
+
+    print("Added files to dataset.");
+    print("See dataset: https://app.viam.com/data/datasets?id=$datasetId");
+
+    viamClient.close();
+    return 0;
+
+  } catch (error) {
+    print("Error creating dataset. It may already exist.");
+    print("See: https://app.viam.com/data/datasets");
+    print("Exception: $error");
+    viamClient.close();
+    return 1;
+  }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Capture, annotate, and add images to a dataset
```

Comment:
Depending on where the structure of these pages lands, maybe it'd make sense to frame this more like "Combine these steps in a single script" to acknowledge the overlap with other sections